mllib/src/main/scala/org/apache/spark/mllib/linalg/Matrices.scala

```diff
@@ -34,6 +34,12 @@ sealed trait Matrix extends Serializable {
   /** Number of columns. */
   def numCols: Int
 
+  /** Flag that keeps track whether the matrix is transposed or not. False by default. */
+  private[mllib] var isTrans = false
```

Comment:
We can make `isTranspose` a `val`/`def`. Add make a new constructor that takes `isTransposed` as input.